### PR TITLE
Ignore world state when rendering debug worlds

### LINKED ISSUE #2069: Sodium is weird with some blocks (desynchronized)
### Bug Description

Sodium is weird with some blocks

image
https://cdn.discordapp.com/attachments/1148716397610946623/1150450982103285820/2023-09-10_17-21-08.mp4

### Reproduction Steps

Enter the world of debugging
Try to open/close trapdoors

### Log File

[latest.log](https://github.com/CaffeineMC/sodium-fabric/files/12568571/latest.log)

### Crash Report

[crash-2023-09-10_17.32.28-client.txt](https://github.com/CaffeineMC/sodium-fabric/files/12568570/crash-2023-09-10_17.32.28-client.txt)

### PR DESCRIPTION
**Pull request by embeddedt**
Fixes #2069 by mimicking vanilla's very hacky approach of using the debug chunk generator directly inside `getBlockState` calls. To avoid polluting the rest of the code with `isDebugWorld` checks, we fabricate a fake block container that contains the same blocks a debug world would. This should be indistinguishable from vanilla at runtime, while not having any measurable impact on performance for regular worlds.

### CODE REVIEW

File: src/main/java/me/jellysquid/mods/sodium/client/world/cloned/ClonedChunkSection.java [hunk 1]
```diff
@@ -71,6 +75,34 @@ public ClonedChunkSection(World world, WorldChunk chunk, @Nullable ChunkSection
...
 
+    /**
+     * Construct a fake PalettedContainer whose contents match those of the debug world. This is needed to
+     * match vanilla's odd approach of short-circuiting getBlockState calls inside its render region class.
+     */
+    @NotNull
+    private static PalettedContainer<BlockState> constructDebugWorldContainer(ChunkSectionPos pos) {
+        var container = new PalettedContainer<>(Block.STATE_IDS, Blocks.AIR.getDefaultState(), PalettedContainer.PaletteProvider.BLOCK_STATE);
+        // We use swapUnsafe to avoid going through the lock when we own the container
+        for (int y = 0; y < 16; y++) {
```
Comment (jellysquid3) [inline src/main/java/me/jellysquid/mods/sodium/client/world/cloned/ClonedChunkSection.java]: Can we avoid iterating over every Y-coordinate here? The code only cares about Y-level 60 and 70.

File: src/main/java/me/jellysquid/mods/sodium/client/world/cloned/ClonedChunkSection.java [hunk 2]
```diff
@@ -71,6 +75,34 @@ public ClonedChunkSection(World world, WorldChunk chunk, @Nullable ChunkSection
...
+            int worldY = ChunkSectionPos.getOffsetPos(pos.getY(), y);
+            if (worldY == 60) {
+                BlockState state = Blocks.BARRIER.getDefaultState();
+                for (int z = 0; z < 16; z++) {
+                    for (int x = 0; x < 16; x++) {
+                        container.swapUnsafe(x, y, z, state);
```
Comment (jellysquid3) [inline src/main/java/me/jellysquid/mods/sodium/client/world/cloned/ClonedChunkSection.java]: Why use `PalettedContainer.swapUnsafe`? We don't care about the existing value. We can just use `PalettedContainer.set(int, Object)` instead.

**Review by jellysquid3 — CHANGES_REQUESTED** [review]
I hate this workaround, but understand that it's a necessary evil. While it is not likely to be called often in practice, I would like to clean up the code slightly.

File: src/main/java/me/jellysquid/mods/sodium/client/world/cloned/ClonedChunkSection.java (hunk 2 above)
Comment (embeddedt) [reply src/main/java/me/jellysquid/mods/sodium/client/world/cloned/ClonedChunkSection.java]: `set` acquires and releases a lock on each call.

Comment (jellysquid3) [reply src/main/java/me/jellysquid/mods/sodium/client/world/cloned/ClonedChunkSection.java]: Ah... that method is not public. Oh well.

**Review by jellysquid3 — APPROVED** [review]

**Pull request merged**